WeakAuras/Prototypes.lua

```diff
@@ -27,6 +28,31 @@ function WeakAuras.GetRange(unit, checkVisible)
   return LibRangeCheck:GetRange(unit, checkVisible);
 end
 
+local LibClassicCast
+if WeakAuras.IsClassic() then
+  LibClassicCast = LibStub("LibClassicCast-1.0")
+end
+
+function WeakAuras.UnitCastingInfo(unit)
```

Comment:
I prefer the check to be outside of the function so:

  if not WeakAuras.IsClassic() then
    WeakAuras.UnitCastingInfo = UnitCastingInfo
 else
  WeakAuras.UnitCastingInfo =  function(unit)
  if UnitIsUnit(unit, "player") then
    return CastingInfo()
  else
    return LibClassicCast:UnitCastingInfo(unit)
  end
end